src/test/java/com/example/demo/SecurityIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+
+        testUser1 = new AppUser(null, "testName", "testUsername", "testPassword", new ArrayList<>());
+        testUser2 = new AppUser(null, "testName", "testUsername", "testPassword", new ArrayList<>());
+        // these guys will need accountIds
+
+        userService.saveUser(testUser1);
+        userService.saveUser(testUser2);
+    }
+
+    @Test
+    public void itShouldNotAllowLoginWithNoCredentials() throws Exception {
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.post("/auth/login"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+    }
+
+    @Test
+    public void itShouldNotAllowLoginWithBadCredentials() throws Exception {
+        String requestBody = TestUtils.asJsonString(new AuthRequest("badUsername", "badPassword"));
+
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.post("/auth/login")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithoutToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithBadlyFormattedToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid)
+                                .header("Authorization", "Bearer badtoken"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldNotAllowPrivateAccessWithOldToken() throws Exception {
+        String testUser1Accountid = "GET THIS ONCE USERS HAVE ACCOUNTIDS";
+        MvcResult result = mvc.perform(
+                        MockMvcRequestBuilders.get("/orderbook/buy/" + testUser1Accountid)
+                                .header("Authorization", "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9 .eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ .SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void itShouldAllowAccessToPublicBuyOrderbook() throws Exception {
```

Comment:
Is this a duplicate of the test in the MatcherControllerTest?

Reply:
well, I'm a bit confused about where my tests should lie. In the integration testing bit I want to, in particular, check that the security layer is doing its job, whereas in the MatcherControllerTest suite I'm checking that the MatcherController is doing its job. I don't know if this is reasonable thinking and I reckon it might warrant a proper discussion